src/MainWindow.cpp

```diff
@@ -330,11 +335,15 @@ void MainWindow::init()
     connect(resetLayoutAction, &QAction::triggered, [=]() {
             restoreState(defaultWindowState);
             restoreOpenTabs(defaultOpenTabs);
+            ui->viewDBToolbarAction->setChecked(!ui->toolbarDB->isHidden());
+            ui->viewExtraDBToolbarAction->setChecked(!ui->toolbarExtraDB->isHidden());
+            ui->viewProjectToolbarAction->setChecked(!ui->toolbarProject->isHidden());
         });
     QAction* simplifyLayoutAction = layoutMenu->addAction(tr("Simplify Window Layout"));
     simplifyLayoutAction->setShortcut(QKeySequence(tr("Shift+Alt+0")));
     connect(simplifyLayoutAction, &QAction::triggered, [=]() {
             toggleTabVisible(ui->pragmas, false);
+            viewMenuPragmas->setChecked(false);
```

Comment:
These two lines can be replaced by a simple call to `viewMenuPragmas->activate()`

Reply:
Umm.. I've tried the code according to this review. The `active()` requires a parameter of type `QAction::ActionEvent`. However, `QAction::ActionEvent` doesn't have suitable anything for this case. Could you give me a little more advice?

Reply:
For this case you have to use `QAction::Trigger` and it will be as if the user had pressed the button. There is also a parameterless slot called `trigger()`.